op-node/rollup/derive/engine_queue.go

```diff
@@ -221,10 +233,31 @@ func (eq *EngineQueue) SafeL2Head() eth.L2BlockRef {
 	return eq.safeHead
 }
 
+func (eq *EngineQueue) EngineSyncTarget() eth.L2BlockRef {
+	return eq.engineSyncTarget
+}
+
+// Determine if the engine is syncing to the target block
+func (eq *EngineQueue) isEngineSyncing() bool {
+	return eq.unsafeHead.Hash != eq.engineSyncTarget.Hash
+}
+
 func (eq *EngineQueue) Step(ctx context.Context) error {
 	if eq.needForkchoiceUpdate {
 		return eq.tryUpdateEngine(ctx)
 	}
+	// Trying unsafe payload should be done before safe attributes
+	// It allows the unsafe head can move forward while the long-range consolidation is in progress.
+	if eq.unsafePayloads.Len() > 0 {
+		if err := eq.tryNextUnsafePayload(ctx); err != io.EOF {
+			return err
+		}
+		// EOF error means we can't process the next unsafe payload. Then we should process next safe attributes.
+	}
```

Comment:
```suggestion
        unsafePayloadEOF := false
	// It allows the unsafe head can move forward while the long-range consolidation is in progress.
	if eq.unsafePayloads.Len() > 0 {
		if err := eq.tryNextUnsafePayload(ctx); err != io.EOF {
			return err
		} else {
			unsafePayloadEOF = true
		}
		// EOF error means we can't process the next unsafe payload. Then we should process next safe attributes.
	}
```


If `err == io.EOF`, `tryNextUnsafePayload()` method is blocked for waiting new unsafe payloads. Suggestion to not ignore condition.

After safe block logic is done, we reach 
```go
	if outOfData || unsafePayloadEOF {
		return io.EOF
	} else {
		return 
	}
```

Reply:
After offline discussion, we do not need this change. The code will eventually return `io.EOF` when reached upper code segment.